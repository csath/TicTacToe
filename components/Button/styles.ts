import { StyleSheet } from "react-native";
import Colors from "../../constants/Colors";

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.btnBackground,
    borderRadius: 6,
    justifyContent: "center",
    alignItems: "center",
    paddingVertical: 12,
    paddingHorizontal: 30,
    marginBottom: 15,
  },
  linkContainer: {
    justifyContent: "center",
    alignItems: "center",
    paddingVertical: 12,
    paddingHorizontal: 15,
  },
  title: {
    color: Colors.btnText,
    fontSize: 18,
  },
  linkTitle: {
    color: Colors.btnTextSecondary,
    fontSize: 18,
  },
});

export default styles;
