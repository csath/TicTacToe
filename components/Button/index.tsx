import React from "react";
import {
  Pressable,
  PressableProps,
  Text as DefaultText,
  Text,
} from "react-native";
import styles from "./styles";

type CustomButtonProps = {
  variant?: "Button" | "Link";
  text: string;
  textStyle?: DefaultText["props"]["style"];
} & PressableProps;

const CustomButton = ({
  text,
  variant,
  textStyle,
  ...props
}: CustomButtonProps) => {
  return (
    <Pressable
      style={({ pressed }: { pressed: boolean }) => [
        {
          opacity: pressed ? 0.5 : 1,
        },
        variant === "Link" ? styles.linkContainer : styles.container,
      ]}
      {...props}
    >
      <Text
        style={[
          variant === "Link" ? styles.linkTitle : styles.title,
          textStyle,
        ]}
      >
        {text}
      </Text>
    </Pressable>
  );
};

export default CustomButton;
