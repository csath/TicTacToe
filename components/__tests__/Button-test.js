import * as React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';

it(`Button renders correctly`, () => {
  const tree = renderer.create(<Button text='test'/>).toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree.children.length).toBe(1);
});
