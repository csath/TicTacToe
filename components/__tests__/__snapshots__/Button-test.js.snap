// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "opacity": 1,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 6,
        "justifyContent": "center",
        "marginBottom": 15,
        "paddingHorizontal": 30,
        "paddingVertical": 12,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontSize": 18,
        },
        undefined,
      ]
    }
  >
    test
  </Text>
</View>
`;
